hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALPrettyPrinter.java

```diff
@@ -274,7 +289,8 @@ public void processFile(final Configuration conf, final Path p)
         Map<String, Object> txn = key.toStringMap();
         long writeTime = key.getWriteTime();
         // check output filters
-        if (table != null && !((TableName) txn.get("table")).toString().equals(table)) {
+        if (!tableList.isEmpty() &&
+          !tableList.contains(((TableName) txn.get("table")).toString())) {
```

Comment:
Looks like this efficient search with `contains()` is the reason behind using `Set` instead of `List`.
Btw, `(TableName)` is redundant cast here.